### ISSUE TITLE
probably don't use global features (arch/os/format) to optimize rule matching

### ISSUE THREAD

**Issue by williballenthin** [mandiant/capa#931]
we currently optimize rule matching by downselecting from all rules to the rules that could potentially match, using the presence of features referenced in those rules. we probably don't want to use arch/os/format features to do this downselection. i dont know if we want to hardcode this, or if there's some other policy way of doing it.

**Comment by mr-tz** [comment]
one idea: move `arch`, `os`, etc. to the `meta` section? most rules use them under a top-level `and` IIRC

pro:
- quickly filter out rules

con:
- duplication in some abstraction rules (negligible, I think)

**Comment by williballenthin** [comment]
we support these global features so we can do things like:

```yaml
- or:
  - and:
    - arch: i386
    - offset: 4
  - and:
    - arch: amd64
    - offset: 8        
```
which i think is pretty elegant. this was discussed in https://github.com/mandiant/capa/discussions/701

**Comment by williballenthin** [comment]
oh, we already have this helper: https://github.com/mandiant/capa/blob/fe5d88585ce85a7823770bdde7078635611fcb69/capa/features/common.py#L444-L449

<img width="518" alt="image" src="https://user-images.githubusercontent.com/156560/160678955-67b29390-0276-4aa7-8237-b491e5bfe962.png">


which should be enough

**Comment by mr-tz** [comment]
ah, can't we do `offset/x32` only `number/x32`...?

**Comment by williballenthin** [comment]
actually, both of those work, but we can't do anything similar with OS (im not sure we've ever needed to). i wonder if we should remove the `/x32` syntax with preference to the block syntax, though its a concise shorthand.

https://github.com/mandiant/capa/blob/fe5d88585ce85a7823770bdde7078635611fcb69/capa/rules.py#L241-L254

**Comment by williballenthin** [comment]
not using global features during downselect is implemented in c1b28f58d0e535da4ea6126a0bed7ea011e814cc
